go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -439,6 +439,12 @@ func transformRoutePlan(ctx *plancontext.PlanningContext, op *operators.Route) (
 
 	switch stmt := stmt.(type) {
 	case sqlparser.SelectStatement:
+		if op.Lock != sqlparser.NoLock {
+			stmt.SetLock(op.Lock)
+		}
+		if op.Comments != nil {
```

Comment:
comments apply to all statement types and not just to SELECT and UNION, right?

Reply:
yes, I only removed setMiscFunc from the select plan, that's why it is used only here. Let me check for other statement types and if I can remove them from these places as well.

Reply:
In DML operator transform, the original AST is provided which contains the comment
```go
func buildDML(op OpWithAST, qb *queryBuilder) {
	qb.stmt = op.Statement()
	qb.dmlOperator = op
}
```